provider/pkg/client/client.go

```diff
@@ -152,3 +165,39 @@ func (c *clientImpl) Delete(ctx context.Context, typeName, identifier string) er
 
 	return err
 }
+
+// Wraps c.api.GetResource with Not Found error retry logic to try to compensate for eventual consistency issues for
+// newly created resources.
+func (c *clientImpl) getResourceRetryNotFound(
+	ctx context.Context,
+	typeName, identifier string,
+) (map[string]interface{}, error) {
+	retryBackoff := retry.NewExponentialJitterBackoff(30 * time.Second)
```

Comment:
I'd reduce the max duration here a bit. We'd now wait up to ~60 seconds. That seems a bit too long to me.

Eventual consistency delays for those services are usually in the millis.

Reply:
I'll send a quick follow-up PR, this one merged already.